### PR TITLE
chore: Add mockable http client to simple client

### PR DESCRIPTION
**Pull request by epociask**
## Fixes Issue

Fixes #

### CODE REVIEW

File: client/simple.go [hunk 1]
```diff
@@ -10,12 +10,15 @@ import (
...
 var (
 	// 503 error type informing rollup to failover to other DA location
-	ErrServiceUnavailable = fmt.Errorf("eigenda service is unavailable")
+	ErrServiceUnavailable = fmt.Errorf("eigenda service is temporarily unavailable")
 )
 
-// TODO: Add support for custom http client option
 type Config struct {
-	URL string
+	URL string // EigenDA proxy RPC URL
```
Comment (samlaf) [inline client/simple.go:17]: nit: it's actually a REST-ish (not quite kek) endpoint, not RPC

File: client/simple.go [hunk 2]
```diff
@@ -10,12 +10,15 @@ import (
...
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
```
Comment (samlaf) [inline client/simple.go:21]: Is this really needed here? How are you mocking the client? with some mock library?
Can we use https://pkg.go.dev/net/http/httptest#Server.Client instead to simplify the code and not have this interface (don't think its typical to have an interface for an http client in golang, but not 100% sure)

File: client/simple.go [hunk 3]
```diff
@@ -80,7 +99,7 @@ func (c *SimpleCommitmentClient) GetData(ctx context.Context, comm []byte) ([]by
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("received error response, code=%d, msg = %s", resp.StatusCode, string(b))
+		return nil, fmt.Errorf("received error response when reading from eigenda, code=%d, msg = %s", resp.StatusCode, string(b))
```
Comment (samlaf) [inline client/simple.go:102]: we don't know that the error comes from eigenda right? it could come from the proxy itself, like some 500 or whatever

Comment (samlaf) [reply client/simple.go:102]: maybe just change to "from eigenda-proxy"?

File: client/simple.go [hunk 4]
```diff
@@ -106,16 +125,17 @@ func (c *SimpleCommitmentClient) SetData(ctx context.Context, b []byte) ([]byte,
...
 	}
 
+	// failover signal
 	if resp.StatusCode == http.StatusServiceUnavailable {
 		return nil, ErrServiceUnavailable
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("failed to store data: %v, err = %s", resp.StatusCode, string(b))
+		return nil, fmt.Errorf("received error response when dispersing to eigenda, code=%d, err = %s", resp.StatusCode, string(b))
```
Comment (samlaf) [inline client/simple.go:134]: same here

File: client/simple.go (hunk 2 above)
Comment (epociask) [reply client/simple.go:21]: its easiest for me to just bind it to an interface for mocking - done this across quite a few projects in the past

File: client/simple.go (hunk 4 above)
Comment (epociask) [reply client/simple.go]: [a2e80ee](https://github.com/Layr-Labs/eigenda-proxy/pull/209/commits/a2e80ee0dd07827280a4383a2753acb8f4e4169e)

File: client/simple.go (hunk 1 above)
Comment (epociask) [reply client/simple.go]: [a2e80ee](https://github.com/Layr-Labs/eigenda-proxy/pull/209/commits/a2e80ee0dd07827280a4383a2753acb8f4e4169e)

File: client/simple.go (hunk 3 above)
Comment (epociask) [reply client/simple.go]: [a2e80ee](https://github.com/Layr-Labs/eigenda-proxy/pull/209/commits/a2e80ee0dd07827280a4383a2753acb8f4e4169e)

**Review by samlaf — APPROVED** [review]

**Pull request merged**